rts/motoko-rts/src/gc/generational/remembered_set.rs

```diff
@@ -0,0 +1,280 @@
+//! Remembered set.
+//! Serves for recording pointer locations trapped by the write barrier.
+//!
+//! Hash-set implementation. Linked-list collision handling.
+//!
+//! Hash function = (ptr / WORD_SIZE) % TABLE_SIZE
+//!
+//! Hash-table (length N):
+//! ----------------------------------
+//! | entry[0]   | collision_ptr[0]   |
+//! ----------------------------------
+//! | entry[1]   | collision_ptr[1]   |
+//! ----------------------------------
+//! |  ...                           |
+//! ----------------------------------
+//! | entry[N-1] | collision_ptr[N-1] |
+//! ----------------------------------
+//!
+//! Per collision a new linked list node is appended:
+//!
+//!                               Collision node
+//!                        ------------------------------
+//! prev_collision_ptr --> | entry | next_collision_ptr |
+//!                        ------------------------------
+//!
+//! Amortized hash-table growth when exceeding a defined threshold.
```

Comment:
Good idea, I also thought of that. Fallback to full collection in the case of too many remembered set entries. I currently tested the case but because GC is anyway often running out of message limit, it was not (yet) a problem. Probably more a topic for later optimization once DTS is available and as a precaution for a probably relative pathological case. Another remedy would be to schedule young GC when it grows to big. But this would require call stack root set analysis. I keep it on my todo list for the moment.